e2e-tests/tfm/cypress/e2e/journeys/utilisation-reports/sort-premium-payments.spec.js

```diff
@@ -0,0 +1,117 @@
+
+  describe('when status column heading is clicked', () => {
+    it('should sort the rows by status', () => {
+      pages.utilisationReportPage.premiumPaymentsTab.premiumPaymentsTable.header.status().click();
+
+      // Check that the rows are sorted by status by checking the facility IDs.
+
+      // Facility 22222222 has status 'DOES_NOT_MATCH' (alphabetically first) and has the lowest total reported payments
+      // (secondary sorting criteria, ascending), so it should appear first.
+      cy.assertText(pages.utilisationReportPage.premiumPaymentsTab.premiumPaymentsTable.rows().eq(0).find('th').first(), '22222222');
+
+      // Facility 11111111 also has status 'DOES_NOT_MATCH' but has a slightly higher total reported payments, so it
+      // should appear second.
+      cy.assertText(pages.utilisationReportPage.premiumPaymentsTab.premiumPaymentsTable.rows().eq(1).find('th').first(), '11111111');
+
+      // Facility 33333333 has status 'TO_DO' (alphabetically last), so it should appear last.
+      cy.assertText(pages.utilisationReportPage.premiumPaymentsTab.premiumPaymentsTable.rows().eq(2).find('th').first(), '33333333');
+    });
+  });
+
+  describe('when total payment received column heading is clicked', () => {
+    it('should sort the rows by total payment received', () => {
+      pages.utilisationReportPage.premiumPaymentsTab.premiumPaymentsTable.header.totalPaymentsReceived().click();
+
+      // Check that the rows are sorted by total payment received by checking the facility IDs.
+
+      // Facility 33333333 has no reported payments, so it should appear first.
+      cy.assertText(pages.utilisationReportPage.premiumPaymentsTab.premiumPaymentsTable.rows().eq(0).find('th').first(), '33333333');
+
+      // Facility 22222222 has the highest total reported payments (200 EUR), so it should appear second.
+      cy.assertText(pages.utilisationReportPage.premiumPaymentsTab.premiumPaymentsTable.rows().eq(1).find('th').first(), '22222222');
+
+      // Facility 11111111 has the lowest total reported payments (300 USD), so it should appear last.
+      cy.assertText(pages.utilisationReportPage.premiumPaymentsTab.premiumPaymentsTable.rows().eq(2).find('th').first(), '11111111');
+    });
+  });
+
+  describe('when total reported payments column heading is clicked', () => {
+    it('should sort the rows by total reported payments', () => {
+      pages.utilisationReportPage.premiumPaymentsTab.premiumPaymentsTable.header.totalReportedPayments().click();
+
+      // Check that the rows are sorted by total reported payments by checking the facility IDs.
+
+      // NOTE: This column is the default sorting column and defaults to ascending, therefore the descending case is being tested.
+
+      // Facility 11111111 has reported payment currency 'EUR' (alphabetically first), so it should appear first.
+      cy.assertText(pages.utilisationReportPage.premiumPaymentsTab.premiumPaymentsTable.rows().eq(0).find('th').first(), '11111111');
```

Comment:
Could we move the selectors into `premiumPaymentsTable` please, keep the actual test simpler and DRY? Not sure about the naming/structure but would like it to be e.g:

```js
cy.assertText(aSelector, '11111111');
cy.assertText(anotherSelector, '22222222');
```

Reply:
Thanks, pulled out some of this into: ...`premiumPaymentsTable.facilityIdByRowIndex(0)`

Is this similar to what you had in mind/okay? Thanks